azure/services/managedclusters/spec.go

```diff
@@ -417,6 +484,52 @@ func computeDiffOfNormalizedClusters(managedCluster containerservice.ManagedClus
 		}
 	}
 
+	// TODO: Not sure this comparison is correct since defaults are set at AKS level, do checks
```

Comment:
I'll keep looking at this tomorrow, but I think this might need to be split into checks for each field unfortunately. I tried creating a cluster without any autoscaler settings, changing one of them, and changing it back. The first change went as expected, but the second update seemed like it got the controller caught in a loop trying to update it but it was never taking effect. I'll dig in more tomorrow to figure out exactly what's going on in that scenario.

Reply:
yeah, tested setting the entire block of options and it syncs fine but with some missing it doesn't reconcile as I had hoped. It's looking like I'd need to set the default values for all settings if any value is set in the CR but I don't really like that :/ I'll think on it some more tomorrow.

Reply:
I ended up setting default values for all the settings in the CR. It's not ideal imo, but it's the only way I could think of that handled the scenario when someone set a value but then removed it from the CR.